hilti/toolchain/src/compiler/jit.cc

```diff
@@ -64,99 +80,288 @@ bool CxxCode::save(std::ostream& out) const {
     return ! out.fail();
 }
 
-#ifdef HILTI_HAVE_JIT
-#include <hilti/compiler/detail/clang.h>
+JIT::JIT(std::shared_ptr<Context> context, bool dump_code) : _context(std::move(context)), _dump_code(dump_code) {}
+
+JIT::~JIT() { _finish(); }
+
+hilti::Result<std::shared_ptr<const Library>> JIT::build() {
+    util::timing::Collector _("hilti/jit");
 
-JIT::JIT(std::shared_ptr<Context> context) : _context(std::move(context)) {
-    _jit = std::make_unique<detail::ClangJIT>(_context);
+    if ( auto rc = _checkCompiler(); ! rc )
+        return rc.error();
+
+    if ( auto rc = _initialize(); ! rc )
+        return rc.error();
+
+    if ( auto rc = _compile(); ! rc )
+        return rc.error();
+
+    auto library = _link();
+    _finish(); // clean up no matter if successful
+    return library;
 }
 
-JIT::~JIT() {}
+hilti::Result<Nothing> JIT::_checkCompiler() {
+    auto cxx = hilti::configuration().cxx;
 
-bool JIT::compile() {
-    util::timing::Collector _("hilti/jit/compile");
+    // We ignore the output, just see if running the compiler works. `-dumpversion`
+    // works with both GCC and clang, but unlikely to be supported by something
+    // other than a compiler.
+    if ( auto rc = _spawnJob(cxx, {"-dumpversion"}); ! rc )
```

Comment:
Let's maybe inject our flags here as well to catch incompatibilities early on.

Reply:
Not sure that would work well / help much